test/base/test_mps.jl

```diff
@@ -962,6 +962,9 @@ end
       @test inner(psi', MPO(a, s), psi) ≈ Cpm[i - ss + 1, j - ss + 1]
     end
 
+    # Test sites argument (also regression test for issue #105)
+    Cxz = correlation_matrix(psi, "X", "Z"; sites=2:2:6)
```

Comment:
Can you check the result using `test_correlation_matrix`? It might not handle the `sites` keyword argument properly right now but that should be easy to fix.